cmd/lotus-shed/state-stats.go

```diff
@@ -226,41 +229,70 @@ func loadChainStore(ctx context.Context, repoPath string) (*StoreHandle, error)
 		return nil, xerrors.Errorf("lotus repo doesn't exist")
 	}
 
-	lr, err := r.Lock(repo.FullNode)
+	lkrepo, err := r.Lock(repo.FullNode)
 	if err != nil {
 		return nil, err
 	}
 
-	bs, err := lr.Blockstore(ctx, repo.UniversalBlockstore)
+	cold, err := lkrepo.Blockstore(ctx, repo.UniversalBlockstore)
 	if err != nil {
-		return nil, fmt.Errorf("failed to open blockstore: %w", err)
+		return nil, xerrors.Errorf("failed to open universal blockstore %w", err)
 	}
 
+	path, err := lkrepo.SplitstorePath()
```

Comment:
What is going on here ? What was breaking that made you change this?

Reply:
When you open universal blockstore it should be wrapped around all of the splitstore right?

Reply:
@ZenGround0: when using the original I couldn't get it to run - when it initialised and figured out what the latest head is (from the metadata store) it would then try and load that tipset but fail with not-found errors. Switching to this makes it work. My assumption is that this works fine for a non-splitstore setup, but breaks when using a splitstore, and nobody running splitstore has been using the commands that use this code until I tried it.

Both patterns are found throughout lotus-shed, I copied a lot of this new stuff from elsewhere. I'd like to understand it a bit better to make it consistent throughout, maybe just make a helper function for this. But I haven't taken the time to bother going that deep yet. If you have more of a clue then I'm all ears!

Reply:
The reason I'm concerned is that I am almost certain this used to work with a splitstore setup.  I was running state stat stuff back in 2023 when splitstore hotstore started working again.  Not found errors on current state lookup are also unexpected.  I guess you can merge this since these are only shed commands but it is a scary feeling to accept that the universal blockstore just doesn't work.

Reply:
ok, this is good to know, worth investigating